swap_meet/vendor.py

```diff
@@ -1,2 +1,105 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory == None:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            index_of_item = self.inventory.index(item)
+            self.inventory.pop(index_of_item)
+            return item
+        return False
+    
+    # Returns the item with a matching id
+    def get_by_id(self, item_id):
+        for item in self.inventory:
+            if item.id == item_id:
+                return item
+        return None
+    
+    # This methods removes my_item from my inventory and adds it to my friend's inventory.
+    # Returns True when removes and adds and False when my item is not in my inventory
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            item_to_other_vendor = self.remove(my_item)
+            other_vendor.add(item_to_other_vendor)
+
+            item_from_other_vendor_to_me = other_vendor.remove(their_item)
+            self.add(item_from_other_vendor_to_me)
```

Comment:
I really like that you used the return value of `remove` (`item_to_other_vendor` and `item_from_other_vendor_to_me`)! I think it's a good pattern.

Reply:
Thank you! I have to admit I was super confused throughout the project with the name of parameters and variables but I'm happy that at the end I understood the concepts and usage of these parameters inside my methods.